daemon/start.go

```diff
@@ -158,6 +158,12 @@ func (daemon *Daemon) containerStart(container *container.Container) (err error)
 			container.SetExitCode(126)
 		}
 
+		// attempted to mount a file onto a directory, or a directory onto a file, maybe from user specified bind mounts
+		if strings.Contains(errDesc, syscall.ENOTDIR.Error()) {
+			errDesc += ": check that all specified host paths exist and are the correct file type for all volumes"
```

Comment:
`correct file type` may be a bit ambiguous, perhaps we should mention exactly what the issue is (mounting a directory on a directory); Just some thoughts (but not very happy with it :smile:);

```
are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists.
```